core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -104,17 +103,25 @@ public BlockInfo getInfo(long blockId) throws IOException {
   }
 
   /**
-   * @return the info of all active block workers
+   * @return the info of all available block workers
    */
-  public List<BlockWorkerInfo> getWorkerInfoList() throws IOException {
-    List<BlockWorkerInfo> infoList = new ArrayList<>();
+  public List<BlockWorkerInfo> getAvailableWorkers() throws IOException {
+    return getAllWorkers().stream()
+        // Filter out unavailable workers.
+        .filter(w -> w.getNetAddress().getTieredIdentity()
+              .strictTiersMatch(TieredIdentityFactory.getInstance()))
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * @return the info of all block workers
+   */
+  public List<BlockWorkerInfo> getAllWorkers() throws IOException {
```

Comment:
Isn't this `getAvailableWorkers`? Also, should it be private?

Reply:
We use it in the `LogLevel` command for finding all workers in the cluster